lib/ueberauth/strategy/linkedin.ex

```diff
@@ -124,30 +116,29 @@ defmodule Ueberauth.Strategy.LinkedIn do
     }
   end
 
-  defp skip_url_encode_option, do: [path_encode_fun: fn(a) -> a end]
-
-  defp user_query do
-    "/v1/people/~:(id,picture-url,email-address,firstName,lastName)?format=json"
-  end
+  defp skip_url_encode_option, do: [path_encode_fun: &(&1)]
 
   defp fetch_user(conn, token) do
     conn = put_private(conn, :linkedin_token, token)
-    resp = Ueberauth.Strategy.LinkedIn.OAuth.get(token, user_query, [], skip_url_encode_option)
+    resp = Ueberauth.Strategy.LinkedIn.OAuth.get(token, @user_query, [],
+      skip_url_encode_option())
 
-IO.puts("linkedin fetch_user, resp = #{inspect resp}")
+    IO.puts("linkedin fetch_user, resp = #{inspect resp}")
 
     case resp do
-      { :ok, %OAuth2.Response{status_code: 401, body: _body}} ->
+      {:ok, %OAuth2.Response{status_code: 401, body: _body}} ->
         set_errors!(conn, [error("token", "unauthorized")])
-      { :ok, %OAuth2.Response{status_code: status_code, body: user} }
+      {:ok, %OAuth2.Response{status_code: status_code, body: user}}
         when status_code in 200..399 ->
           put_private(conn, :linkedin_user, user)
-      { :error, %OAuth2.Error{reason: reason} } ->
+      {:error, %OAuth2.Error{reason: reason}} ->
         set_errors!(conn, [error("OAuth2", reason)])
     end
   end
 
   defp option(conn, key) do
-    Dict.get(options(conn), key, Dict.get(default_options, key))
+    conn
+    |> options
+    |> Dict.get(key, Dict.get(default_options(), key))
```

Comment:
`Dict.get/2` and `Dict.get/3` have been deprecated. Do you know which functions would be best used here?

Reply:
The whole `Dict` module is in fact deprecated. I think for this application, the closest replacement is `Map`, because it could enforce unicity of the option keys.

Reply:
We should update it so it's compatible with Elixir 2.0 when it comes out in a few years.